packages/jira-adapter/src/change_validators/workflowsV2/status_mappings.ts

```
@@ -0,0 +1,261 @@
+}): Promise<StatusMapping[]> => {
+  const issueTypeReferences = await getWorkflowSchemeMigrationIssueType({
+    workflowSchemeInstance,
+    workflowName,
+    elementsSource,
+  })
+  if (_.isEmpty(issueTypeReferences)) {
+    return []
+  }
+  const statusMappings = projectReferences.flatMap(project =>
+    issueTypeReferences.map(issueType =>
+      createStatusMapping({
+        project,
+        issueType,
+        statusesToMigrate: removedStatuses,
+      })))
+  const missingStatusMappings = _.differenceWith(statusMappings, existingStatusMappings, isSameStatusMappings)
+  if (_.isEmpty(missingStatusMappings)) {
+    return []
+  }
+  return statusMappings
+}
+
+const getStatusMappingsFormat = (statusMappings: StatusMapping[]): string =>
+  `statusMappings = [
+    ${statusMappings.map(statusMapping => createStatusMappingStructure(statusMapping)).join('\n    ')}
+]`
+
+export const workflowStatusMappingsValidator: ChangeValidator = async (changes, elementsSource) => {
+  const relevantChanges = getRelevantChanges(changes)
+  if (elementsSource === undefined || _.isEmpty(relevantChanges)) {
+    return []
+  }
+  const projects = await getInstancesFromElementSource(elementsSource, [PROJECT_TYPE])
+  // all the active workflowSchemes
+  const workflowSchemeNameToInstance = await awu(projects)
+    .filter(projectHasWorkflowSchemeReference)
+    .map(project => project.value.workflowScheme.getResolvedValue(elementsSource))
+    .filter(isInstanceElement)
+    .keyBy(instance => instance.elemID.getFullName())
+
+  const workflowSchemeNameToProjectReferences = _.groupBy(
+    projects.filter(projectHasWorkflowSchemeReference)
+      .map(project => new ReferenceExpression(project.elemID, project)),
+    project => project.value.value.workflowScheme.elemID.getFullName(),
+  )
+  return awu(relevantChanges).flatMap(async change => {
+    const workflowInstance = getChangeData(change)
+    if (!isStatusMappings(workflowInstance.value.statusMappings)) {
```

if it's `undefined`, you don't want to return an error here, right?

---

right

---

so this condition isn't enough I guess?

---

when its `undefined`, the user try to do the change without knowing that he needs to add the `statusMappings`. It is probably the common flow that we tell him to stop and supply him with the relevant mappings (undefined pass this validation). 
The schemeGuard is meant to find when the user already adds a `statusMappings` but it is not in the right structure. IIIC the condition is enough

---

if undefined pass this validation so all good 👍